### PR TITLE
add tests for ref port

### PR DESCRIPTION
**Pull request by slemasne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            //  Confirm that the portfolio was successfully created.  Any failures will result in
+            //  a ApiException being thrown which contain the relevant response code and error message
+            Assert.That(ReferencePortfolio.Id.Code, Is.EqualTo(request.Code));
+        }
+
+        [Test]
+
+        public void SetConstituents()
+        {
+
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:70]: remove whitespace

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            var instrument3 = _instrumentIds[2];
+
+            // Set date for the constituents, the date from which the constituents are valid
+            var date = new DateTimeOffset(2020, 1, 1, 0, 0, 0, TimeSpan.Zero);
+
+            // Weighting type
+            var weighting = UpsertReferencePortfolioConstituentsRequest.WeightTypeEnum.Static;
+
+            // Create the request to upsert constituents
+            var UpsertConstituentsRequest = new UpsertReferencePortfolioConstituentsRequest(
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:83]: `upsertConstituentsRequest`

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                     new ReferencePortfolioConstituentRequest(
+                        instrumentIdentifiers: new Dictionary<string, string> {[TestDataUtilities.LusidInstrumentIdentifier] = instrument3},
+                        weight: 40,
+                        currency: "GBP"
+                    ),
+                }
+            );
+
+            // Upsert the constituents into LUSID
+            var SetConstituents = _apiFactory.Api<IReferencePortfolioApi>().UpsertReferencePortfolioConstituents(TutorialScope, PorfolioCode, UpsertConstituentsRequest);
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:117]: `setConstituents`

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+            var RefHoldings = _apiFactory.Api<IReferencePortfolioApi>().GetReferencePortfolioConstituents(TutorialScope, PorfolioCode, date);
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:119]: `constituentHoldings`

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+            Assert.That(RefHoldings.Constituents.Count(), Is.EqualTo(4));
+
+            RefHoldings.Constituents.Sort((h1, h2) => String.Compare(h1.InstrumentUid, h2.InstrumentUid, StringComparison.Ordinal));
+
+            Assert.That(RefHoldings.Constituents[1].InstrumentUid, Is.EqualTo(_instrumentIds[0]));
+            Assert.That(RefHoldings.Constituents[2].InstrumentUid, Is.EqualTo(_instrumentIds[1]));
+            Assert.That(RefHoldings.Constituents[3].InstrumentUid, Is.EqualTo(_instrumentIds[2]));
+
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:128]: remove redundant whitespace

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+using Lusid.Sdk.Model;
+using Lusid.Sdk.Tests.Utilities;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+
+namespace Lusid.Sdk.Tests.Tutorials.Ibor
+{
+
+    [TestFixture]
+    public class ReferencePortfolios
```
[the remark this answers is not in the document]
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:14]: Sure, done.

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+        public void SetUp()
+        {
+            //  Initialise the API factory
+            _apiFactory = LusidApiFactoryBuilder.Build("secrets.json");
+
+            // Load default instruments which will be used in the tests
+            var instrumentsLoader = new InstrumentLoader(_apiFactory);
+            _instrumentIds = instrumentsLoader.LoadInstruments().OrderBy(x => x).ToList();
+
```
[the remark this answers is not in the document]
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:33]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        private ILusidApiFactory _apiFactory;
+        private IList<string> _instrumentIds;
+
+        //  This defines the scope that entities will be created in the test
+        private readonly string PorfolioCode = $"MyPort-{Guid.NewGuid().ToString()}";
+        const string TutorialScope = "Testdemo1";
+
+        [OneTimeSetUp]
+
```
[the remark this answers is not in the document]
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:24]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+            _apiFactory = LusidApiFactoryBuilder.Build("secrets.json");
+
+            // Load default instruments which will be used in the tests
+            var instrumentsLoader = new InstrumentLoader(_apiFactory);
+            _instrumentIds = instrumentsLoader.LoadInstruments().OrderBy(x => x).ToList();
+
+        }
+
+        [Test]
+
```
[the remark this answers is not in the document]
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:37]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        public void CreateReferencePortfolio()
+
+        {
+            //  Create date for the portfolio
+            var create_date = new DateTimeOffset(2000, 1, 1, 0, 0, 0, TimeSpan.Zero);
```
[the remark this answers is not in the document]
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:42]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs (hunk 1 above)
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:70]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs (hunk 2 above)
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:83]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs (hunk 3 above)
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs:117]: done

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs (hunk 5 above)
Comment (slemasne) [reply sdk/Lusid.Sdk.Tests/tutorials/Ibor/ReferencePorfolios.cs]: done

**Pull request merged**